src/components/notifications/GenericErrorToast.tsx

```diff
@@ -14,25 +14,44 @@
  * limitations under the License.
  ***************************************************************************** */
 
-import React from 'react';
+import React, { useState } from 'react';
+import { copyTextToClipboard } from '../../helpers/copyHandler';
+import { createStyleSelector } from '../../helpers/styleCreators';
 import { GenericError } from '../../stores/NotificationsStore';
 
 export default function GenericErrorToast(props: GenericError) {
 	const { description, header, action } = props;
+	const [copied, setCopied] = useState(false);
+	const copyDetailsText = createStyleSelector('toast-action__text', copied ? 'copied' : null);
+
+	const copy = () => {
+		copyTextToClipboard(description);
+		setCopied(true);
+	};
+
+	const shortenedDesc = description.length > 32 ? `${description.slice(0, 33)}...` : description;
```

Comment:
If you want to have 32 characters then slice from 0 to 32, current slice will include 33 characters, because numeration starts at 0